src/plugins/grpc_plugin/grpc_plugin.cc

```diff
@@ -113,8 +116,11 @@ bool GrpcPlugin::Initialize(runtime::core::AimRTCore* core_ptr) noexcept {
     asio_executor_ptr_->Start();
 
     core_ptr_->RegisterHookFunc(runtime::core::AimRTCore::State::kPostShutdown, [this] {
-      asio_executor_ptr_->Shutdown();
-      asio_executor_ptr_->Join();
+      if (asio_executor_ptr_) {  // Check if resource exists to ensure safe release
+        asio_executor_ptr_->Shutdown();
+        asio_executor_ptr_->Join();
+        asio_executor_ptr_.reset();  // Explicitly release the resource
```

Comment:
Defensive programming is good; let's add similar content to the 150-line range to maintain consistency.